common/headtracker/head_tracker.go

```diff
@@ -114,16 +117,15 @@ func (ht *headTracker[HTH, S, ID, BLOCK_HASH]) Start(ctx context.Context) error
 		// anyway when we connect (but we should not rely on this because it is
 		// not specced). If it happens this is fine, and the head will be
 		// ignored as a duplicate.
-		err := ht.handleInitialHead(ctx)
-		if err != nil {
-			if ctx.Err() != nil {
-				return ctx.Err()
+		onSubscribe := func() {
```

Comment:
Why do we create a separate function instead of passing `handlInitialHead` and what does this accomplish inside `ListenForNewHeads` ?

Reply:
I've defined an anonymous function here to capture HeadTracker's context and logger. `handleInitialHead` is part of HeadTracker's initialization, so we should not log errors with headListener's logger. There is an option to achieve this without anonymous functions, but they seem suboptimal. 

We have to pass it to `ListenForNewHeads` to ensure it's called after a new subscription was created to avoid the following race:
1. Fetch initial head (block 10)
2. Chain produces block 11 and stops producing new blocks (that's how SimulatedBacked behaves in large portion of OCR tests)
3. `headListener` subscribes and does not receive new heads.
4. Consumers of HeadTracker's `LatestChain` only see block 10.

I've consistently observed this issue while attempting to make LogPoller use LatestChain. Unfortunately, that attempt failed, as many OCR tests have fragile logic related to head production because of SimulatedBackend. To reduce the scope of the PR, I've decided to leave the direct call for now but keep this change as it's a bug fix.